### ISSUE TITLE
Issue with plot_images

### ISSUE THREAD

**Issue by juliesnilsen** [hyperspy/hyperspy#1328]
Using 

 m = s.get_lines_intensity()
 hs.plot.plot_images(m) 

in the latest update (1.1) produces a value error. Apparently, this is because plot_images has been changed to only allow "Signal2D" objects to be plotted but get_lines_intensity is not returning that object. It can be solved by using m = [si.as_signal2D((0,1)) for si in s.get_lines_intensity()] instead.

When trying to resolve this issue I discovered that the following example about plotting EDS data in the documentation does not work (and should be updated):

si_EDS = hs.load("core_shell.hdf5")
im = si_EDS.get_lines_intensity()
hs.plot.plot_images(hs.transpose(im, signal_axes=2),
tight_layout=True, cmap='RdYlBu_r', axes_decor='off',
colorbar='single', saturated_pixels=2, scalebar='all',
scalebar_color='black', suptitle_fontsize=16,
padding={'top':0.8, 'bottom':0.10, 'left':0.05,
'right':0.85, 'wspace':0.20, 'hspace':0.10})

**Comment by to266** [comment]
Thanks for the report!

while the `as_signal2D` does definitely work, the example with `transpose` just misses one star:

``` python
hs.plot.plot_images(hs.transpose(*im, signal_axes=2),
                    # other arguments
                    )
```

The difference between the two is slightly subtle: `transpose` only changes how the data is viewed / read from memory (e.g. column-first instead of row-first), while `as_signal2D` actually reorders the stored bytes, so it's read in the most efficient manner. This reordering often also leads to a data copy, which is worth considering if it's a large signal.

**Comment by ericpre** [comment]
Fix in #2130.
